### ISSUE TITLE
Sorting by price is broken when using a comma separator

### ISSUE THREAD

**Issue by bknowles** [duffn/theouterrim#61]
Sorting Gear by price, they are displayed with a comma. But this appears to break sorting order, where “100” sorts after “10,500”. Standard numeric sorting is also broken, where 25 sorts after 200.

I think you’re going to have to convert them from strings to integers, then convert them back on display.

Note that if the site is going to be localized into other languages, they might not use a comma as their thousands separator, so you might not want to bake that in.

This might also require breaking out the “(R)” for Restricted into a separate column.

**Comment by bknowles** [comment]
Let me see if I can get you a screenshot:

![D8BF6F71-6500-4408-A83C-826EC0E5F248](https://user-images.githubusercontent.com/1151895/75118108-abe29080-563c-11ea-9dd5-39e0e1d4afd4.png)

**Comment by duffn** [comment]
Yeah, that's very broken, thanks.

**Comment by AndreLemay** [comment]
I'll get started on this, at least the data for now. I'm not sure I want to make any changes to the tables themselves if we're going to merge in the mui-datatables pr soon.

**Comment by duffn** [comment]
Excellent! Thanks for snapping up another issue. As soon as I finished poking around at the new table, we'll get that in.

I look forward to your solution on the `(R)` issue here. Whatever you come up with could also be used in #45 with `+` for some weapons.

**Comment by AndreLemay** [comment]
What I'm thinking so far is that the restricted value will be another hidden column (like generatedId), we'll give the price columns a custom renderer that conditionally prepends (R) on restricted items, and I'm hoping that the sorting will be done based on the data value and not the display value. If not, a custom sort function should handle that.

**Comment by AndreLemay** [comment]
Got this working, though it looks like the table wants to break the price column once you have more than 4 digits in the price, like this:

![image](https://user-images.githubusercontent.com/787013/75300015-7d56e800-57eb-11ea-87bb-2ac6b263ea23.png)

Is this something we care to fix? Should be possible to just change the word-breaking css. I've also noticed that a lot of column seem to break annoyingly leading to some items, like weapons with lots of qualities or anything that's in all the core books, ending up 5-6 rows tall. We could just make nothing break, though that would probably significantly widen most of the tables.

**Comment by duffn** [comment]
Very cool that you got this working.

I think we're fine just leaving the wrapping as it is, considering the dynamism of it. If it ends up being really annoying we can revisit it.

**Comment by AndreLemay** [comment]
Sounds good, once #52 is merged I'll open the PR for this

**Comment by duffn** [comment]
Awesome, merged.
